exporters/ironic.go

```diff
@@ -37,18 +38,24 @@ func NewIronicExporter(config *ExporterConfig, logger log.Logger) (*IronicExport
 		}
 	}
 
+	// NOTE(Sharpz7) Gophercloud V2 adds this new field ResourceBase.
+	// For whatever reason, it adds a v1 field to the URL,
+	// so it sends requests to /v1/v1 if left unfixed.
+	config.ClientV2.ResourceBase = config.ClientV2.Endpoint
+
 	// Set Microversion workaround
-	microversion, err := apiversions.Get(config.Client, "v1").Extract()
+	microversion, err := apiversions.Get(context.TODO(), config.ClientV2, "v1").Extract()
 	if err == nil {
-		exporter.Client.Microversion = microversion.Version
+		config.ClientV2.Microversion = microversion.Version
+		config.Client.Microversion = microversion.Version
```

Comment:
Just curious: what happens if this microversion isn't available? Are we implicitly raising the minimum microversion needed to run the exporter? The other fields we're exporting here appear to have been added in very very early versions of Ironic.

Reply:
I didn't want to change current behaviour, so I just copied. A good note though for the future.

Reply:
My point is; this could potentially change behavior: the fields we are currently pulling are from the earliest versions of Ironic; this change would move the compatibility "line" up something like 4-5 years :D.

@niedbalski Do you have a general policy/practice for this sorta thing?

Reply:
No, wait I think you are right. I will revert back to what it was originally if possible...

Reply:
Yeah, I think we might need to just see the behavior if we request a field in a microversion that does not exist in the underlying cloud. I can help you get a test environment for this up if you want.